service/operator/service.go

```diff
@@ -163,116 +159,331 @@ func (s *Service) addFunc(obj interface{}) {
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
 
-	err := s.addFuncError(obj)
+	err := ProcessCreate(obj, s)
 	if err != nil {
 		s.logger.Log("error", fmt.Sprintf("%#v", err), "event", "create")
 	}
 }
 
-func (s *Service) addFuncError(obj interface{}) error {
-	customObject, ok := obj.(*kvmtpr.CustomObject)
+func (s *Service) deleteFunc(obj interface{}) {
+	s.mutex.Lock()
```

Comment:
Do we need the mutexes because these are run in parallel? I guess a comment explaining this scenario might be useful.